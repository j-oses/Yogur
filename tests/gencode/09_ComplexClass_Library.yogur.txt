 {0}    ssp 28;
 {1}    lda 0 0;
 {2}    ldc 23456;
 {3}    sto;
 {4}    ujp 21; \\ endFun1
\\ def mod
\\ fun1:
 {5}    ssp 7;
 {6}    sep 3;
 {7}    lda 0 0;
 {8}    lda 0 5;
 {9}    ind;
 {10}   lda 0 5;
 {11}   ind;
 {12}   lda 0 6;
 {13}   ind;
 {14}   div;
 {15}   lda 0 6;
 {16}   ind;
 {17}   mul;
 {18}   sub;
 {19}   sto;
 {20}   retf;
\\ endFun1:
 {21}   ujp 40; \\ endFun2
\\ def random
\\ fun2:
 {22}   ssp 5;
 {23}   sep 1;
 {24}   lda 1 0;
 {25}   mst 1;
 {26}   lda 1 0;
 {27}   ind;
 {28}   ldc 15245;
 {29}   mul;
 {30}   ldc 3784;
 {31}   add;
 {32}   ldc 65535;
 {33}   cup 2 5; \\ fun1
 {34}   sto;
 {35}   lda 0 0;
 {36}   lda 1 0;
 {37}   ind;
 {38}   sto;
 {39}   retf;
\\ endFun2:
 {40}   ujp 79; \\ endFun3
\\ def randomize
\\ fun3:
 {41}   ssp 7;
 {42}   sep 4;
 {43}   lda 0 6;
 {44}   ldc 0;
 {45}   sto;
\\ for5:
 {46}   lda 0 6;
 {47}   ind;
 {48}   ldc 2;
 {49}   leq;
 {50}   fjp 70; \\ endFor5
 {51}   lod 0 5;
 {52}   inc 0;
 {53}   lda 0 6;
 {54}   ind;
 {55}   chk 0 2;
 {56}   ixa 1;
 {57}   mst 1;
 {58}   mst 1;
 {59}   cup 0 22; \\ fun2
 {60}   ldc 3;
 {61}   cup 2 5; \\ fun1
 {62}   sto;
 {63}   lda 0 6;
 {64}   lda 0 6;
 {65}   ind;
 {66}   ldc 1;
 {67}   add;
 {68}   sto;
 {69}   ujp 46; \\ for5
\\ endFor5:
 {70}   lod 0 5;
 {71}   inc 3;
 {72}   mst 1;
 {73}   mst 1;
 {74}   cup 0 22; \\ fun2
 {75}   ldc 20;
 {76}   cup 2 5; \\ fun1
 {77}   sto;
 {78}   retp;
\\ endFun3:
 {79}   ujp 118; \\ endFun4
\\ def copy
\\ fun4:
 {80}   ssp 8;
 {81}   sep 4;
 {82}   lda 0 7;
 {83}   ldc 0;
 {84}   sto;
\\ for6:
 {85}   lda 0 7;
 {86}   ind;
 {87}   ldc 2;
 {88}   leq;
 {89}   fjp 111; \\ endFor6
 {90}   lod 0 5;
 {91}   inc 0;
 {92}   lda 0 7;
 {93}   ind;
 {94}   chk 0 2;
 {95}   ixa 1;
 {96}   lod 0 6;
 {97}   inc 0;
 {98}   lda 0 7;
 {99}   ind;
 {100}  chk 0 2;
 {101}  ixa 1;
 {102}  ind;
 {103}  sto;
 {104}  lda 0 7;
 {105}  lda 0 7;
 {106}  ind;
 {107}  ldc 1;
 {108}  add;
 {109}  sto;
 {110}  ujp 85; \\ for6
\\ endFor6:
 {111}  lod 0 5;
 {112}  inc 3;
 {113}  lod 0 6;
 {114}  inc 3;
 {115}  ind;
 {116}  sto;
 {117}  retp;
\\ endFun4:
 {118}  ujp 150; \\ endFun7
\\ def replace
\\ fun7:
 {119}  ssp 19;
 {120}  sep 4;
 {121}  lda 0 18;
 {122}  ldc 0;
 {123}  sto;
\\ for9:
 {124}  lda 0 18;
 {125}  ind;
 {126}  ldc 2;
 {127}  leq;
 {128}  fjp 149; \\ endFor9
 {129}  mst 1;
 {130}  lod 0 5;
 {131}  inc 0;
 {132}  lda 0 18;
 {133}  ind;
 {134}  chk 0 2;
 {135}  ixa 4;
 {136}  lda 0 6;
 {137}  lda 0 18;
 {138}  ind;
 {139}  chk 0 2;
 {140}  ixa 4;
 {141}  cup 2 80; \\ fun4
 {142}  lda 0 18;
 {143}  lda 0 18;
 {144}  ind;
 {145}  ldc 1;
 {146}  add;
 {147}  sto;
 {148}  ujp 124; \\ for9
\\ endFor9:
 {149}  retp;
\\ endFun7:
 {150}  ujp 177; \\ endFun8
\\ def randomize
\\ fun8:
 {151}  ssp 7;
 {152}  sep 4;
 {153}  lda 0 6;
 {154}  ldc 0;
 {155}  sto;
\\ for10:
 {156}  lda 0 6;
 {157}  ind;
 {158}  ldc 2;
 {159}  leq;
 {160}  fjp 176; \\ endFor10
 {161}  mst 1;
 {162}  lod 0 5;
 {163}  inc 0;
 {164}  lda 0 6;
 {165}  ind;
 {166}  chk 0 2;
 {167}  ixa 4;
 {168}  cup 1 41; \\ fun3
 {169}  lda 0 6;
 {170}  lda 0 6;
 {171}  ind;
 {172}  ldc 1;
 {173}  add;
 {174}  sto;
 {175}  ujp 156; \\ for10
\\ endFor10:
 {176}  retp;
\\ endFun8:
 {177}  lda 0 1;
 {178}  inc 12;
 {179}  ldc 22;
 {180}  sto;
 {181}  lda 0 1;
 {182}  inc 13;
 {183}  ldc false;
 {184}  sto;
 {185}  mst 0;
 {186}  lda 0 1;
 {187}  cup 1 151; \\ fun8
 {188}  lda 0 27;
 {189}  ldc 0;
 {190}  sto;
\\ for11:
 {191}  lda 0 27;
 {192}  ind;
 {193}  ldc 2;
 {194}  leq;
 {195}  fjp 210; \\ endFor11
 {196}  mst 0;
 {197}  lda 0 15;
 {198}  lda 0 27;
 {199}  ind;
 {200}  chk 0 2;
 {201}  ixa 4;
 {202}  cup 1 41; \\ fun3
 {203}  lda 0 27;
 {204}  lda 0 27;
 {205}  ind;
 {206}  ldc 1;
 {207}  add;
 {208}  sto;
 {209}  ujp 191; \\ for11
\\ endFor11:
 {210}  mst 0;
 {211}  lda 0 1;
 {212}  cup 1 151; \\ fun8
 {213}  mst 0;
 {214}  lda 0 1;
 {215}  lda 0 15;
 {216}  movs 12;
 {217}  cup 13 119; \\ fun7
 {218}  stp;
