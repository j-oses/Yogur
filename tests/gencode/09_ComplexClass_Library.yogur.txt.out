Run successfully completed
Final P-machine state:
PC = 218   MP = 0   SP = 27   EP = -1   NP = 100   code size = 219
STORE = 
(0,Int 28709)
(1,Int 2)
(2,Int 2)
(3,Int 2)
(4,Int 4)
(5,Int 2)
(6,Int 2)
(7,Int 2)
(8,Int 9)
(9,Int 2)
(10,Int 2)
(11,Int 2)
(12,Int 14)
(13,Int 22)
(14,Bool False)
(15,Int 2)
(16,Int 2)
(17,Int 2)
(18,Int 4)
(19,Int 2)
(20,Int 2)
(21,Int 2)
(22,Int 9)
(23,Int 2)
(24,Int 2)
(25,Int 2)
(26,Int 14)
(27,Int 3)
(28,Bool False)
(29,Int 0)
(30,Int 0)
(31,Int (-1))
(32,Int 218)
(33,Int 1)
(34,Int 2)
(35,Int 2)
(36,Int 2)
(37,Int 4)
(38,Int 2)
(39,Int 2)
(40,Int 2)
(41,Int 9)
(42,Int 2)
(43,Int 2)
(44,Int 2)
(45,Int 14)
(46,Int 3)
(47,Bool False)
(48,Int 2)
(49,Int 1)
(50,Int 54)
(51,Int 142)
(52,Int 9)
(53,Int 42)
(54,Int 3)
(55,Int 12)
(56,Int 14)
(57,Int 1)
(58,Int 34)
(59,Int 190932164)
(60,Int 65535)
(61,Int 54)
(62,Int 28709)
(63,Int 190903455)
(64,Int 65535)
(65,Null)
(66,Null)
(67,Null)
(68,Null)
(69,Null)
(70,Null)
(71,Null)
(72,Null)
(73,Null)
(74,Null)
(75,Null)
(76,Null)
(77,Null)
(78,Null)
(79,Null)
(80,Null)
(81,Null)
(82,Null)
(83,Null)
(84,Null)
(85,Null)
(86,Null)
(87,Null)
(88,Null)
(89,Null)
(90,Null)
(91,Null)
(92,Null)
(93,Null)
(94,Null)
(95,Null)
(96,Null)
(97,Null)
(98,Null)
(99,Null)
